compose/overrides/generate-config-files.py

```diff
@@ -0,0 +1,410 @@
+
+    for key, value in variables.items():
+        parts = key.split('__')
+
+        if len(parts) != 3:
+            logger.error('Error: Variable "%s" does not match PREFIX__SECTION__KEY format', key)
+            sys.exit(1)
+
+        # Spaces in section names are encoded with '0x20' (HEX representation of a space character)
+        # This is inspired by Gitea, which uses 0x2E for a dot character
+        section = parts[1].replace('0x20', ' ')
+        key = parts[2]
+
+        if section not in config:
+            # section does not exist yet
+            config[section] = {}
+
+        config[section][key] = value
+
+    if not os.path.exists(path):
+        logger.info(f'Generating {os.path.basename(path)} since it does not exist yet')
+    else:
+        logger.info(f'Updating {os.path.basename(path)}')
+
+    with open(path, 'w') as file:
+        config.write(file)
+
+def generate_gunicorn_config_file(path: str):
+    # Source: https://github.com/haiwen/seafile-docker/blob/da9bf740e4a093a0c25c4ae9a09e08069194fc73/scripts/scripts_11.0/setup-seafile-mysql.py#L1213
+    config = """
+import os
+
+daemon = True
+workers = 5
+
+# default localhost:8000
+bind = "127.0.0.1:8000"
+
+# Pid
+pids_dir = '/opt/seafile/pids'
+pidfile = os.path.join(pids_dir, 'seahub.pid')
+
+# for file upload, we need a longer timeout value (default is only 30s, too short)
+timeout = 1200
+
+limit_request_line = 8190
+"""
+
+    if not os.path.exists(path):
+        logger.info(f'Generating {os.path.basename(path)} since it does not exist yet')
+    else:
+        logger.info(f'Updating {os.path.basename(path)}')
+
+    with open(path, 'w') as file:
+        # Use lstrip() to remove leading whitespace
+        file.write(config.lstrip())
+
+def generate_seahub_settings_file(path: str):
+    database_config_template = """
+DATABASES = {
+    'default': {
+        'ENGINE': 'django.db.backends.mysql',
+        'NAME': '%(name)s',
+        'USER': '%(username)s',
+        'PASSWORD': '%(password)s',
+        'HOST': '%(host)s',
+        'PORT': '%(port)s',
+        'OPTIONS': {'charset': 'utf8mb4'},
+    }
+}
+"""
+
+    database_config = {
+        'name': 'seahub_db',
+        'username': 'root',
+        'password': os.environ['DB_ROOT_PASSWD'],
+        'host': os.environ['DB_HOST'],
+        'port': '3306',
+    }
+
+    cache_config_template = """
+CACHES = {
+    'default': {
+        'BACKEND': '%(backend)s',
+        'LOCATION': '%(host)s:%(port)s',
+    },
+    'locmem': {
+        'BACKEND': 'django.core.cache.backends.locmem.LocMemCache',
+    },
+}
+"""
+
+    # Should use memcached as default cache backend
+    cache_backend = os.environ.get('SEAHUB__CACHE_BACKEND', 'memcached')
+    if cache_backend == 'memcached':
+        django_cache_backend = 'django_pylibmc.memcached.PyLibMCCache'
+    elif cache_backend == 'redis':
+        # TODO: The redis python package is missing from the container image, therefore the redis cache backend does not work!
```

Comment:
I've tested redis here: https://github.com/simonhammes/seafile-release/compare/config...simonhammes:seafile-release:config-check-redis?expand=1 (Diff)

- The redis python package is still missing, even in 11.0.12
- The warning re: the invalid configuration inside seafile.conf is not triggered, I'm guessing this happens because I'm not testing against S3
  - Assumption: Seafile does not cache anything if files are stored on the local filesystem